Grbl_Esp32/tools/SpindleClass.h

```diff
@@ -0,0 +1,126 @@
+/*
+    SpindleClass.h
+
+    Header file for a Spindle Class
+    See SpindleClass.cpp for more details
+
+    Part of Grbl_ESP32
+
+    2020 -	Bart Dring This file was modified for use on the ESP32
+                    CPU. Do not use this with Grbl for atMega328P
+
+    Grbl is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    Grbl is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with Grbl.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+#include "grbl.h"
+#include <driver/dac.h>
+
+
+#define SPINDLE_STATE_DISABLE  0  // Must be zero.
+#define SPINDLE_STATE_CW       bit(0)
+#define SPINDLE_STATE_CCW      bit(1)
+
+#define UNDEFINED_PIN 255  // Pins will be set to this if not defined
+
+#ifndef SPINDLE_CLASS_H
+#define SPINDLE_CLASS_H
+
+// This is the base class. Do not use this as your spindle
+class Spindle {
+  public:
+    virtual void init(); // not in constructor because this also gets called when $$ settings change
+    virtual float set_rpm(float rpm);
+    virtual void set_pwm(uint32_t duty);
```

Comment:
Maybe this should be omitted, so set_pwm() is not exposed.

Reply:
done